### PR TITLE
Spring cloud azure refactor global properties

### PR DESCRIPTION
**Pull request by saragluna**
Refactor the AzureGlobalProperties to allow configuring http/amqp related properties in the uppermost level.

### CODE REVIEW

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/context/AzureGlobalConfigurationEnvironmentPostProcessor.java [hunk 1]
```diff
@@ -91,11 +105,30 @@ public int getOrder() {
             this.springPropertyName = "spring.cloud.azure." + springPropertyName;
             this.converter = converter;
         }
+
+        private static Function<String, String> convertMillisToDuration() {
```
Comment (stliu) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/context/AzureGlobalConfigurationEnvironmentPostProcessor.java:109]: I think here can just use duration, no need to convert it to string again?

**Review by stliu — APPROVED** [review]
for the rest of the changes I think it is good to merge

could you try not to convert the duration back to string, IIRC you can set duration object into spring's environment and it is smart enough to figure it out

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/context/AzureGlobalConfigurationEnvironmentPostProcessor.java (hunk 1 above)
Comment (saragluna) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/context/AzureGlobalConfigurationEnvironmentPostProcessor.java]: done

**Comment by saragluna** [comment]
@stliu please take a look at the last commit, it fixed a bug in AzureProfile.Environment. I assume when using a known azure environment, it should not allow modification.

**Review by stliu — APPROVED** [review]

**Pull request merged**